### ISSUE TITLE
[Tailwind CSS] Allow to pull colors from your own tailwind.config.js

### ISSUE THREAD

**Issue by alessandro-newzoo** [raycast/extensions#7372]
### Extension

https://www.raycast.com/vimtor/tailwindcss

### Description

Hello, and thank you so much for this fantastic extension!

It will be really great if there was a way to define your own colors, since I'm sure a lot of people have adjusted these to match their branding guidelines, hence rendering this functionality useless for this use case.

I'm not even sure if it's possible, but just thought I'd throw it out there :)

Thank you!

### Who will benefit from this feature?

Whoever is not using the default TailwindCSS color palette.

**Comment by vimtor** [comment]
Hey @alessandro-newzoo thanks for the suggestion! I like the idea but I'm not sure if there is an easy way for the user to add their own colors. Maybe copy pasting the   `colors` field in the `tailwind.config.js`?

**Comment by alessandro-newzoo** [comment]
hey @vimtor !
I'm actually not sure what can be done with Raycast's API, but I'd say one of these options might work:

1. let the user paste the `colors` object into some input box in Raycast, _or_
2. let the user upload their `tailwind.config.js`, _or_
3. let the user paste the path of a locally stored `tailwind.config.js`

My workaround currently is to clone your repo, replace `/tailwind.config.js`  with mine, and make these edits to `/src/search-colors.tsx`:

```diff
- import colors from "tailwindcss/colors";
+ import config from '../tailwind.config';
+ const colors = config.theme.extend.colors;
```

though this implies that the config file is using `extend` and also that the `colors` object contains no functions, or else it won't work.

So, long story short, probably a field to paste in the `colors` object would be the easiest?

I'm also opening another issue for some more suggestions, I hope you don't mind! ❤️

Thank you,
Alessandro

**Comment by vimtor** [comment]
Hi @alessandro-newzoo!

I have been experimenting with allowing the user to select their own file. After trying different `tailwind.config.js` files from the internet, I concluded that it would be better to create a separate extension to manage color palettes and search within them. There were many edge cases since it is Javascript. Some files were simple, while others imported things or looped through colors. I could consider adding an input to let the user add their own colors, but there is no easy way for Tailwind CSS users to get all the colors in a specific format.

If you don't mind, I will close this issue. If you think more people could benefit, I could create the color palette extension I mentioned.

@raycastbot close this issue.

**Comment by alessandro-newzoo** [comment]
Hi @vimtor !

That's alright, what you say makes sense, I didn't think about people having different structures in their `tailwind.config.js`. 

I will continue using a fork of your extension in which I've added all my colors.

Thanks a lot for this and keep up the great work!
